File: IT/Funzioni/C++/lambda.cpp
/*
file lambda.cpp
compilare con: g++ -std=c++11 lambda.cpp
*/

#include <iostream>

using std::cout;
using std::endl;

int main() 
{
	int n = 3;

	// dichiara una funzione lambda (anonima)
	auto f = [](int n) { return n * n; };

	int square = f(n);

	cout << n << " al quadrato = " << square << endl;
}

File: IT/1/Funzioni/Java/Lambda.java
/* file Lambda.java */

import java.util.function.IntUnaryOperator;

public class Lambda {
	public static void main(String [] args) {
		// dichiara una funzione lambda (anonima)
		IntUnaryOperator f = (n) -> n * n;

		int n = 3;
		int square = f.applyAsInt(n);

		System.out.println(n + " al quadrato = " + square);
	}
}

